### PR TITLE
Fixed issue in file_storage_sparse

### PR DESCRIPTION
**Pull request by jimmyaxod**
Fixed issue in file_storage_sparse partial WriteAt on last block not extending to Size().
The bug meant that a read() / merge was not being done. Just a write().
This meant that previous data on the block could be lost and corrupted.

### CODE REVIEW

File: pkg/storage/modules/copy_on_write_test.go [hunk 1]
```diff
@@ -149,3 +151,74 @@ func TestCopyOnWriteReadOverrunNonMultiple(t *testing.T) {
...
+	blockSize := 64 * 1024
+
+	fstore, err := sources.NewFileStorageSparseCreate("test_data_sparse", uint64(size), blockSize)
+	assert.NoError(t, err)
+
+	t.Cleanup(func() {
+		os.Remove("test_data_sparse")
+	})
+
+	//	fstore := sources.NewMemoryStorage(size)
```
Comment (lgfa29) [inline pkg/storage/modules/copy_on_write_test.go:168]:
```suggestion
```
It seems like a leftover?

**Review by lgfa29 — APPROVED** [review]
LGTM! Just left some minor comments.

It would be interesting to investigate if fuzzing could help us test for these types of edge cases.

File: pkg/storage/modules/copy_on_write_test.go [hunk 2]
```diff
@@ -149,3 +151,74 @@ func TestCopyOnWriteReadOverrunNonMultiple(t *testing.T) {
...
 }
+
+func TestCopyOnWriteCRCIssue(t *testing.T) {
+
+	// Create a new block storage, backed by memory storage
+	size := 16 * 1024
+	blockSize := 64 * 1024
+
+	fstore, err := sources.NewFileStorageSparseCreate("test_data_sparse", uint64(size), blockSize)
+	assert.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (jimmyaxod) [reply pkg/storage/modules/copy_on_write_test.go:162]: Good point. I'll switch Silo tests to use require where appropriate in a separate PR.

File: pkg/storage/modules/copy_on_write_test.go (hunk 1 above)
Comment (jimmyaxod) [reply pkg/storage/modules/copy_on_write_test.go]: 454758083ba76bb024b7b6a8d91aba7d2df057aa
